config/cmake/HDFUseFortran.cmake

```diff
@@ -31,19 +31,27 @@ endif ()
 # Detect name mangling convention used between Fortran and C
 #-----------------------------------------------------------------------------
 include (FortranCInterface)
+
+FortranCInterface_VERIFY()
+
 FortranCInterface_HEADER (
     ${CMAKE_BINARY_DIR}/FCMangle.h
     MACRO_NAMESPACE "H5_FC_"
     SYMBOL_NAMESPACE "H5_FC_"
 )
 
 file (STRINGS ${CMAKE_BINARY_DIR}/FCMangle.h CONTENTS REGEX "H5_FC_GLOBAL\\(.*,.*\\) +(.*)")
-string (REGEX MATCH "H5_FC_GLOBAL\\(.*,.*\\) +(.*)" RESULT ${CONTENTS})
-set (H5_FC_FUNC "H5_FC_FUNC(name,NAME) ${CMAKE_MATCH_1}")
+if (CONTENTS)
+  message(${CONTENTS})
+  string (REGEX MATCH "H5_FC_GLOBAL\\(.*,.*\\) +(.*)" RESULT  ${CONTENTS})
+  set (H5_FC_FUNC "H5_FC_FUNC(name,NAME) ${CMAKE_MATCH_1}")
+endif()
```

Comment:
should there be an else block? (and below, too)  If neither REGEX is found won't H5_FC_FUNC be undefined?

Reply:
FortranCInterface_VERIFY() should fail for else case.

Reply:
Then why is the if needed?

Reply:
Because I wasn't sure what's the right approach:

1. Use F..._VERIFY()
2. Safeguard REGEX with if ().

E.g., approach 2. was used in many places in HDF5 CMake codes.

Reply:
Add a comment to the FortranCInterface_VERIFY() call that it will fail if the compilers are incompatible the remove the if/endifs as they are just confusing.